### DISCUSSION TITLE
posts/2024/24010537/index

### DISCUSSION THREAD

**Discussion by giscus** [GZTimeWalker/blog-comments#24] [Announcements]
# posts/2024/24010537/index

之前给 QNAP NAS 配置 ACME 自动续签总出问题，今天心血来潮搞了一下，其实也不是什么大问题，因此记录一下。

https://blog.gzti.me/posts/2024/24010537/index.html

**Comment by acotor** [comment]
按照文章内容操作都成功了。测试使用 `certbot.key` SSH 登录 nas 如果出现 `load key error in libcrypto`，可以将 `acme_key` 以文件复制或上传的方式放到指定目录再改名改权限，因为文本编辑器复制粘贴容易出问题~

想问一下：
1. 容器中 crontab 自带的定时任务是不是(1)检查--config-home目录下的域名证书是否过期(2)过期则续签？
2. 续签的时候还需要`CF_Token`等参数吗？因为似乎没有保存这些参数的地方。
3. 续签后 deploy 的动作是如何被自动调用的？因为`acme.sh --deploy -d your.example.com --deploy-hook ssh`似乎看起来只是一次性操作？

**Reply by GZTimeWalker** [reply]
以上三个问题均能在 acme.sh 文档中找到解答，crontab 配置就是定时检查，相关参数会被保存在 acme 配置文件夹中，deploy 会在每次签署成功后自动执行。

**Reply by acotor** [reply]
感谢！我就是懒了点哈哈，回头我再好好去看看文档！之前一直用自签名证书自己去信任根证书；后来为了用 QuObject 对象存储，由于反代对象存储老出问题，于是改为手动上传 LE 的证书。这下自动续签方便多了！

**Comment by wcchuang0323** [comment]
請問一下都順利操作完成了，關閉SSH後對自動續簽會有影響嗎?

**Reply by GZTimeWalker** [reply]
关闭 SSH 后将无法进行服务重启操作，证书将不会更新。

**Reply by wcchuang0323** [reply]
了解，謝謝!
